File: Ruby_version/Napakalaki/lib/death_bad_consequence.rb
# encoding: utf-8
# begin death_bad_consequence.rb

require_relative 'numeric_bad_consequence'

module NapakalakiGame
  class DeathBadConsequence < NumericBadConsequence
    
    def initialize(t)
      super(t, Player.MAXLEVEL, @@MAXTREASURES, @@MAXTREASURES)
    end
    
    def to_s
      "\nDescripción: #{@text}\n Estás muerto.\n"
    end
    
    def malRolloString
      "\n Muerte: pierdes todos los tesoros."
    end
    
  end
end

# end death_bad_consequence.rb

File: Java_version/Napakalaki/src/NapakalakiGame/DeathBadConsequence.java
// begin DeathBadConsequence.java
package NapakalakiGame;

public class DeathBadConsequence extends NumericBadConsequence {
    
    public DeathBadConsequence(String t) {
        super(t, Player.MAXLEVEL, NumericBadConsequence.MAXTREASURES, 
                    NumericBadConsequence.MAXTREASURES);
    }
    
    @Override
    public String toString() {
      return super.toString() + "\nEstás Muerto.";
    }
}

// begin DeathBadConsequence.java